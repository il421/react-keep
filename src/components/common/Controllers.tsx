import React from "react";
import { AlignItems, JustifyContent } from "../../common";
import { FlexBox, IconButton } from "../ui-components";
import "../../styles/components/common/_controllers.scss";
import { NoteType } from "../notes";

interface Controllers extends Partial<Pick<HTMLElement, "className">> {
  isMobile?: boolean;
  openDialog: (type: NoteType) => void;
}

export const Controllers: React.FunctionComponent<Controllers> = ({
  className,
  isMobile = false,
  openDialog,
}) => {
  const handleOnClick = (type: NoteType) => () => openDialog(type);

  return (
    <div
      className={`controllers  ${
        isMobile && "controllers--mobile"
      } ${className}`}
    >
      <FlexBox
        vertical={isMobile}
        justifyContent={JustifyContent.spaceBetween}
        alignItems={AlignItems.center}
      >
        <IconButton
          className="controllers__text"
          onButtonClick={handleOnClick(NoteType.text)}
          icon="align-left"
          size="2x"
        />

        <IconButton
          className="controllers__list"
          onButtonClick={handleOnClick(NoteType.list)}
          icon="check-square"
          size="2x"
        />

        <IconButton
          className="controllers__image"
          onButtonClick={handleOnClick(NoteType.image)}
          icon="image"
          size="2x"
        />
      </FlexBox>
    </div>
  );
};
