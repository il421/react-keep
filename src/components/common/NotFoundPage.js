import React from 'react';
import { Link } from 'react-router-dom';

const NotFoundPage = () => (
  <div className="not-found">
    <div className="not-found__wrapper">
      <div className="not-found__text">
        Sorry, such a page does not exist in the App. <br /><br />
        Try to find something interesting on the Dashboard page.
      </div>
      <Link to="/" className="not-found__link button">Keep me ...</Link>
    </div>
  </div>
);

export default NotFoundPage;
