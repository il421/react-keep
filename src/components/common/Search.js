import React from 'react';
import { connect } from 'react-redux';
import { setSearchFilter } from '../../actions/filters';

export const Search = ({ setSearchFilter }) => {

  return (
    <>
      <input
        className="search"
        placeholder="Search"
        onInput={ (evt) => setSearchFilter(evt.target.value) }
      />
    </>
  );
};

const mapDispatchToProps = (dispatch) => {
  return {
    setSearchFilter: (value) => dispatch(setSearchFilter(value))
  };
};

export default connect(undefined, mapDispatchToProps)(Search);
