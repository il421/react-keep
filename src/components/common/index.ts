export * from "./Controllers";
export * from "./DashboardPage";
export * from "./Header";
export * from "./NotFoundPage";
export * from "./Search";
export * from "./SideBar";
