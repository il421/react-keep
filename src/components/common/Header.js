import React from 'react';
import { connect } from 'react-redux';

import { startLogout } from '../../actions/auth';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';

import Search from './Search';

export const Header = ({ startLogout, auth, showSidebar }) => {

  return (
    <header className="header" onClick={(evt) => evt.stopPropagation()}>
      <div className="content-container">
        <div className="header__content">
          <div className="header__sidebar">

            <button className="button button--sidenav" onClick={() => showSidebar()}>
              <FontAwesomeIcon icon="bars" size="2x" />
            </button>

          </div>
          <div>
            <h1 className="header__title">Keep Me</h1>
          </div>
          <div className="header__search">
            <Search />
          </div>
          <div className="header__user-box user-box">
            <div className="user-box__photo">
              <img src={ auth.url } width={ 50 } height={ 50 } />
            </div>
            <div className="user-box__name show-for-mobile">{ auth.name }</div>
            <button className="user-box__logout button button--sidenav" onClick={ startLogout }>
              <FontAwesomeIcon icon="sign-out-alt" size="lg" />
            </button>
          </div>
        </div>
      </div>
    </header>
  );
};


const mapDispatchToProps = (dispatch) => ({
  startLogout: () => dispatch(startLogout())
});

const mapStateToProps = (state) => {
  return {
    auth: state.auth,
    url: state.url,
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(Header);
