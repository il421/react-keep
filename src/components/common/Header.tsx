import React from "react";
import { connect } from "react-redux";
import { useHistory } from "react-router-dom";
import { stringify } from "query-string";
import { AuthStoreState, Store } from "../../store/store.types";
import { QueryKeys, RouteActions } from "../../routers/Routing";
import { startLogout } from "../../actions/auth";
import Search from "./Search";
import { Controllers } from "./Controllers";
import {
  IconButton,
  UserPhoto,
  ContentContainer,
  LinkButton
} from "../ui-components";
import "../../styles/components/header/_header.scss";
import "../../styles/components/header/_user-box.scss";
import "../../styles/components/common/_visibility.scss";
import { ThunkDispatch } from "redux-thunk";

interface HeaderProps {
  showSidebar: (value: boolean) => void;
}

interface StateProps {
  auth: AuthStoreState;
}

interface DispatchProps {
  startLogout: () => void;
}

type Props = HeaderProps & DispatchProps & StateProps;

const Header: React.FunctionComponent<Props> = ({
  startLogout,
  auth,
  showSidebar
}) => {
  const history = useHistory();

  const onClickHandler = (key: keyof typeof QueryKeys) => {
    const query = stringify({
      [key]: RouteActions.edit
    });
    history.push(`${history.location.pathname}?${query}`);
  };

  return (
    <header className="header" onClick={evt => evt.stopPropagation()}>
      <ContentContainer>
        <div className="header__content">
          <IconButton
            className="header__bar"
            icon="bars"
            size="2x"
            onClick={() => showSidebar(true)}
          />

          <h1 className="header__title">Keep Me</h1>

          <Search wrapperClass="header__search" />

          <Controllers className="hide-for-mobile" />

          <div className="header__user-box user-box">
            {auth.name && auth.url && (
              <UserPhoto
                className="user-box__photo"
                src={auth.url}
                height={50}
                width={50}
              />
            )}
            <LinkButton
              text={auth.name ?? "Update profile"}
              type="button"
              className="user-box__link"
              onClick={() => onClickHandler(QueryKeys.user)}
            />
            <IconButton
              className="user-box__logout"
              icon="sign-out-alt"
              size="lg"
              onClick={startLogout}
            />
          </div>
        </div>
      </ContentContainer>
    </header>
  );
};

const mapDispatchToProps = (
  dispatch: ThunkDispatch<{}, {}, any>
): DispatchProps => ({
  startLogout: () => dispatch(startLogout())
});

const mapStateToProps = (state: Store): StateProps => {
  return {
    auth: state.auth
  };
};

export default connect<StateProps, DispatchProps, HeaderProps, Store>(
  mapStateToProps,
  mapDispatchToProps
)(Header);
