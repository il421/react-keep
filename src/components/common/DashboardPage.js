import React, { useState } from 'react';

import NoteSelection from '../note/NoteSelection';
import NotesList from '../note/NotesList';
import Header from './Header';
import SideBar from './SideBar';


const DashboardPage = () => {
  let [showBar, toggleSidebarDisplay] = useState(false);

  const showSidebar = () => {
    toggleSidebarDisplay(showBar = true);
  };

  const hideSidebar = () => {
    toggleSidebarDisplay(showBar = false);
  };

  return (
    <div className="dashboard">
      <Header showSidebar={ showSidebar } />
      <NoteSelection className="content-container"/>
      <NotesList/>

      <SideBar showBar={ showBar } hideSidebar={ hideSidebar } />
    </div>
  );
};
export default DashboardPage;
