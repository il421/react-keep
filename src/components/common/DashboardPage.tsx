import React from "react";
import { ToastContainer } from "react-toastify";
import { History, Path } from "history";
import { JustifyContent } from "../../common";
import { Header } from "./Header";
import { SideBar } from "./SideBar";
import { Controllers } from "./Controllers";
import UserFormModal from "../login/UserFormModal";
import "../../styles/components/common/_dashboard.scss";
import "../../styles/components/notes/_note-modal.scss";
import { QueryKeys } from "../../routers/Routing";
import { stringify } from "query-string";
import {
  NoteType,
  ImageNoteFormModal,
  TextNoteFormModal,
  ListNoteFormModal,
} from "../notes";
import { FlexBox } from "../ui-components";
import { NotesList } from "../notes/NotesList";

export interface DashboardPageProps {
  history: History;
}

export const onNoteSelected = (options: {
  type: NoteType;
  id: string;
  pathname: string;
  push: (path: Path) => void;
}) => {
  let key: keyof typeof QueryKeys;
  const { type, id, pathname, push } = options;
  switch (type) {
    case NoteType.text:
      key = QueryKeys.text;
      break;
    case NoteType.list:
      key = QueryKeys.list;
      break;
    default:
      key = QueryKeys.image;
  }

  const query = stringify({
    [key]: id,
  });
  push(`${pathname}?${query}`);
};

export const DashboardPage: React.FunctionComponent<DashboardPageProps> = ({
  history,
}): JSX.Element => {
  return (
    <FlexBox
      vertical={true}
      justifyContent={JustifyContent.spaceBetween}
      className="dashboard"
    >
      <Header history={history} />
      <NotesList
        onNoteSelected={(type: NoteType, id: string) =>
          onNoteSelected({
            type,
            id,
            pathname: history.location.pathname,
            push: history.push,
          })
        }
      />

      <SideBar />
      <TextNoteFormModal history={history} />
      <ListNoteFormModal history={history} />
      <ImageNoteFormModal history={history} />
      <UserFormModal />
      <Controllers
        isMobile
        openDialog={(query) =>
          history.push(`${history.location.pathname}?${query}`)
        }
      />

      <ToastContainer autoClose={2000} />
    </FlexBox>
  );
};
