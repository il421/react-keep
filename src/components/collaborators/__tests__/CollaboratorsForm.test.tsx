import React from "react";
import { mount, ReactWrapper } from "enzyme";
import {
  CollaboratorForm,
  Props,
  getValidationErrors,
} from "../CollaboratorForm";
import { ValidationErrors } from "final-form";
import { Errors } from "../../../common";
import { collaborators } from "../../../testData/users";
import { defaultAuthState } from "../../../actions/__tests__/auth.test";

let wrapper: ReactWrapper | undefined, props: Props;

beforeEach(() => {
  props = {
    collaborators,
    auth: defaultAuthState.auth,
    addCollaborator: jest.fn(),
  };
  wrapper = mount(<CollaboratorForm {...props} />);
});

afterEach(() => {
  if (wrapper) {
    wrapper.unmount();
    wrapper = undefined;
  }
});

describe("Rendering", () => {
  test("should render CollaboratorForm correctly", () => {
    if (wrapper) {
      expect(wrapper.debug()).toMatchSnapshot();
    }
  });
});

describe("Validation", () => {
  test("should return no errors", () => {
    const noErrors: ValidationErrors = getValidationErrors(
      props.auth.email!,
      collaborators.map((c) => c.email!)
    )({
      email: "test@test.com",
    });
    expect(Object.keys(noErrors).length).toBe(0);
  });

  test("should return value required", () => {
    const requiredError: ValidationErrors = getValidationErrors(
      props.auth.email!,
      collaborators.map((c) => c.email!)
    )({
      email: "",
    });
    expect(Object.keys(requiredError).length).toBe(1);
    expect(requiredError.email).toBe(Errors.required);
  });

  test("should return not email", () => {
    const requiredError: ValidationErrors = getValidationErrors(
      props.auth.email!,
      collaborators.map((c) => c.email!)
    )({
      email: "test",
    });
    expect(Object.keys(requiredError).length).toBe(1);
    expect(requiredError.email).toBe(Errors.email);
  });

  test("should return currentEmail", () => {
    const requiredError: ValidationErrors = getValidationErrors(
      props.auth.email!,
      collaborators.map((c) => c.email!)
    )({
      email: props.auth.email!,
    });
    expect(Object.keys(requiredError).length).toBe(1);
    expect(requiredError.email).toBe(Errors.currentEmail);
  });

  test("should return existing collaborator", () => {
    const requiredError: ValidationErrors = getValidationErrors(
      props.auth.email!,
      ["test@test.com"]
    )({
      email: "test@test.com",
    });
    expect(Object.keys(requiredError).length).toBe(1);
    expect(requiredError.email).toBe(Errors.existingCollaborator);
  });
});
