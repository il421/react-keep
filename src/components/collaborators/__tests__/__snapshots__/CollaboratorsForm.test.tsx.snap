// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rendering should render CollaboratorForm correctly 1`] = `
"<WithFontAwesome>
  <CollaboratorForm collaborators={{...}} auth={{...}} addCollaborator={[Function: mockConstructor]}>
    <BaseForm initialValues={{...}} onSubmit={[Function]} formClassName=\\"collaborators-form\\" validate={[Function]} resetAfterSubmitting={true} getFormActions={[Function]}>
      <ReactFinalForm initialValues={{...}} onSubmit={[Function]} validate={[Function]} mutators={{...}}>
        <form onSubmit={[Function: handleSubmit]} className=\\"collaborators-form\\" autoComplete=\\"off\\">
          <TextInputField name=\\"email\\" type=\\"text\\" placeholder=\\"Email\\" className=\\"collaborators-form__field\\" autoFocus={true}>
            <Field name=\\"email\\" type=\\"text\\" placeholder=\\"Email\\" className=\\"collaborators-form__field\\" autoFocus={true} component={[Function: FieldAdapter]} parse={[Function: parse]}>
              <FieldAdapter input={{...}} meta={{...}} placeholder=\\"Email\\" className=\\"collaborators-form__field\\" autoFocus={true} render={[undefined]}>
                <div className=\\"collaborators-form__field\\">
                  <input name=\\"email\\" value=\\"\\" checked={[undefined]} onBlur={[Function]} onChange={[Function]} onFocus={[Function]} type=\\"text\\" placeholder=\\"Email\\" render={[undefined]} onInput={[Function: handleOnChange]} onPaste={[undefined]} autoComplete=\\"new-password\\" autoFocus={true} />
                </div>
              </FieldAdapter>
            </Field>
          </TextInputField>
          <div>
            <IconButton className=\\"collaborators-form__submit\\" icon=\\"plus-circle\\" size=\\"1x\\" type=\\"submit\\" disabled={true}>
              <button id={[undefined]} className=\\"icon-button collaborators-form__submit \\" onClick={[undefined]} disabled={true} type=\\"submit\\">
                <FontAwesomeIcon icon=\\"plus-circle\\" size=\\"1x\\" border={false} className=\\"\\" mask={{...}} fixedWidth={false} inverse={false} flip={{...}} listItem={false} pull={{...}} pulse={false} rotation={{...}} spin={false} symbol={false} title=\\"\\" transform={{...}} swapOpacity={false}>
                  <svg aria-hidden=\\"true\\" focusable=\\"false\\" data-prefix=\\"fas\\" data-icon=\\"plus-circle\\" className=\\"svg-inline--fa fa-plus-circle fa-w-16 fa-1x \\" role=\\"img\\" xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 512 512\\" style={{...}}>
                    <path fill=\\"currentColor\\" d=\\"M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm144 276c0 6.6-5.4 12-12 12h-92v92c0 6.6-5.4 12-12 12h-56c-6.6 0-12-5.4-12-12v-92h-92c-6.6 0-12-5.4-12-12v-56c0-6.6 5.4-12 12-12h92v-92c0-6.6 5.4-12 12-12h56c6.6 0 12 5.4 12 12v92h92c6.6 0 12 5.4 12 12v56z\\" style={{...}} />
                  </svg>
                </FontAwesomeIcon>
              </button>
            </IconButton>
          </div>
        </form>
      </ReactFinalForm>
    </BaseForm>
  </CollaboratorForm>
</WithFontAwesome>"
`;
